Makefile

```diff
@@ -139,6 +139,14 @@ python-image:
 push-python-image:
 	docker push ${IMAGE_PREFIX}python:${TEST_INFRA_VERSION}
 
+# Build the Csharp build image
+csharp-image:
+	docker build -t ${INIT_IMAGE_PREFIX}build/csharp:${TEST_INFRA_VERSION} containers/init/build/csharp
+
+# Push the cshsarp build image to a docker registry
+push-python-image:
+  docker push ${IMAGE_PREFIX}csharp:${TEST_INFRA_VERSION}
+
```

Comment:
I think there's a typo here. 

```suggestion
# Push the C# build image to a docker registry
push-csharp-image:
  docker push ${INIT_IMAGE_PREFIX}build/csharp:${TEST_INFRA_VERSION}

```

We may want to create a ${BUILD_IMAGE_PREFIX} = ${INIT_IMAGE_PREFIX}/build/ environment variable at the top. That may be a little cleaner and less prone to errors. What do you think?